### PR TITLE
Add report template id to create web application report in Qualys

### PR DESCRIPTION
**Pull request by mathuriga**
## Purpose
Currently, when we generate the Qulays scan, the report doesn't have the required data findings. This improvement uses the report template id to create a report. This template id can be provided from user. Template id should already exist in Qualys application.

## Goals
Introduce a new parameter to get the template id from the user. If the user doesn't provide template-id, default template id will be taken for report creation.

## Approach
Introduce a new parameter to get the template id from the user. If the user doesn't provide template-id, default template id will be taken for report creation.

## User stories
N/A

## Release note
N/A

## Documentation
N/A

## Training
N/A

## Certification
N/A

## Marketing
N/A

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples
N/A

## Related PRs
N/A

## Migrations (if applicable)
N/A

## Test environment
N/A
 
## Learning
N/A

### CODE REVIEW

File: internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/QualysScannerConstants.java [hunk 1]
```diff
@@ -37,6 +37,8 @@
     public static final String REPORT_UPLOAD_RETRY_INTERVAL = "scan_report_upload_retry_interval";
     public static final String SCHEDULER_DELAY = "schedulerDelay";
     public static final String SCAN_URL = "scanURL";
+    public static final String DEFAULT_REPORT_TEMPLATE_ID = "default_report_template_id";
+    public static final String REPORT_TEMPLATE_ID = "reportTemplateID";
```
Comment (ayomawdb) [inline internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/QualysScannerConstants.java]: It looks wrong that we use two format here (camel-case and underscores).

IMO, we should name `REPORT_TEMPLATE_ID` as `PARAMETER_REPORT_TEMPLATE_ID` to show that it's a HTTP param. That will reduce the confusion.

File: internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/service/QualysScanner.java [hunk 2]
```diff
@@ -207,6 +212,17 @@ private Boolean isValidParameters(ScannerScanRequest scannerScanRequest) throws
             scanContext.setProgressiveScanning(parameterMap.get(QualysScannerConstants.PROGRESSIVE_SCAN).get(0));
         }
 
+        // Validate report template ID
```
Comment (ayomawdb) [inline internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/service/QualysScanner.java]:
```suggestion
        // Validate report template ID.
```

File: internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/service/QualysScanner.java [hunk 3]
```diff
@@ -207,6 +212,17 @@ private Boolean isValidParameters(ScannerScanRequest scannerScanRequest) throws
...
+        if (StringUtils.isEmpty(parameterMap.get(QualysScannerConstants.REPORT_TEMPLATE_ID).get(0))) {
+            scanContext.setReportTemplateId(QualysScannerConfiguration.getInstance().getDefaultReportTemplateID());
+            String logMessage =
+                    "Report template ID for the scan is not provided. Default report template ID" + " is set for "
```
Comment (ayomawdb) [inline internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/service/QualysScanner.java]: Did we miss something here. After the "+"

Comment (mathuriga) [reply internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/service/QualysScanner.java]: Resolved

File: internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/service/QualysScanner.java [hunk 4]
```diff
@@ -207,6 +212,17 @@ private Boolean isValidParameters(ScannerScanRequest scannerScanRequest) throws
...
 
+        // Validate report template ID.
+        if (StringUtils.isEmpty(parameterMap.get(QualysScannerConstants.PARAMETER_REPORT_TEMPLATE_ID).get(0))) {
+            scanContext.setReportTemplateId(QualysScannerConfiguration.getInstance().getDefaultReportTemplateID());
+            String logMessage =
+                    "Report template ID for the scan is not provided. Default report template ID is set for "
+                            + scannerScanRequest.getAppId();
+            log.info(new CallbackLog(scanContext.getJobID(), logMessage));
+        } else {
+            scanContext.setReportTemplateId(parameterMap.get(QualysScannerConstants.PARAMETER_REPORT_TEMPLATE_ID).get(0));
```
Comment (ayomawdb) [inline internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/service/QualysScanner.java]: This line seems longer than allowed.

**Pull request merged**